src/services/chain/messages/evm/transfer.ts

```diff
@@ -55,7 +57,7 @@ export class EvmTransferMessage implements ChainMessage<SendToolParams> {
   ];
 
   inProgressComponent() {
-    return InProgressTxDisplay;
+    return this.hasValidWallet ? InProgressTxDisplay : ConnectWalletPrompt;
```

Comment:
Before, we were showing the in progress component immediately, regardless of whether any validation checks had passed. And that was useful to prompt new users to confirm their metamask connection...sort of.

Now, we can determine if the user is making a transfer and they haven't connected a wallet yet, show them a useful prompt to connect metamask.

Then, if that connected wallet passes validation, show the in progress component.


https://github.com/user-attachments/assets/bb9452b8-8346-4ba9-bbd9-2686d5988434

If the wallet doesn't pass the check, never show in progress

https://github.com/user-attachments/assets/4b3c9c17-5f69-4bbb-a027-624b90ce14aa